File: ResourceManager/ViewModels/PopupViewModelBase.vb
Imports Microsoft.Extensions.DependencyInjection
Imports VBNetCore.Mvvm.Abstractions
Imports VBNetCore.Mvvm.Controls
Imports VBNetCore.Mvvm.ViewModels

Namespace ResourceManager.ViewModels
    Public Class PopupViewModelBase
        Inherits PopupViewModel
        Protected ReadOnly Property popupManager As IPopupManager
            Get
                Return Application.ServiceProvider.GetRequiredService(Of IPopupManager)()
            End Get
        End Property
        Protected ReadOnly Property windowManager As IWindowManager
            Get
                Return Application.ServiceProvider.GetRequiredService(Of IWindowManager)()
            End Get
        End Property
        Protected ReadOnly Property mainVM As MainViewModel
            Get
                Return Application.ServiceProvider.GetRequiredService(Of MainViewModel)()
            End Get
        End Property
        Protected ReadOnly Property UiExecution As IUiExecution
            Get
                Return Application.ServiceProvider.GetRequiredService(Of IUiExecution)()
            End Get
        End Property
    End Class
End Namespace

File: ResourceManager/ViewModels/PopupViewModelBase.cs
using Microsoft.Extensions.DependencyInjection;
using NetCore.Mvvm.Abstractions;
using NetCore.Mvvm.Controls;
using NetCore.Mvvm.ViewModels;
using System;

namespace ResourceManager.ViewModels
{
    public class PopupViewModelBase  : PopupViewModel
    {
        protected IPopupManager popupManager => App.ServiceProvider.GetRequiredService<IPopupManager>();
        protected IWindowManager windowManager => App.ServiceProvider.GetRequiredService<IWindowManager>();
        protected MainViewModel mainVM => App.ServiceProvider.GetRequiredService<MainViewModel>();
        protected IUiExecution UiExecution => App.ServiceProvider.GetRequiredService<IUiExecution>();
    }
}
